packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts

```diff
@@ -246,4 +247,22 @@ describe('Diagnostics', () => {
     const fileUri = getFileUri(workspacePath, filePath, defaultErrorPath);
     expect(fileUri).to.equal(defaultErrorPath);
   });
+
+  it('Should build the absolute file path when constructing the fileUri', () => {
+    const filePath = 'src/classes/Testing.cls';
+    const asoluteFilePath = getAbsoluteFilePath(filePath, workspacePath);
+    expect(asoluteFilePath).to.equal(workspacePath + '/' + filePath);
+  });
+
+  it('Should not duplicate the workspace path when filePath is already absolute', () => {
+    const filePath = `${workspacePath}/src/classes/Testing.cls`;
+    const asoluteFilePath = getAbsoluteFilePath(filePath, workspacePath);
+    expect(asoluteFilePath).to.equal(filePath);
+  });
+
+  it('Should use the workspace path as fileUri when filePath is undefined', () => {
+    const filePath = undefined;
+    const asoluteFilePath = getAbsoluteFilePath(filePath, workspacePath);
+    expect(asoluteFilePath).to.equal(workspacePath);
+  });
```

Comment:
I checked out the failing test and I'd just update it to the following: 
```
 const [fileDesc, reportedError] = setDiagnosticsStub.getCall(
            index
          ).args;
          const expectedFile = vscode.Uri.file(
            getAbsoluteFilePath(row.filePath, getRootWorkspacePath())
          );
          expect(fileDesc).to.deep.equal(expectedFile);
          expect(reportedError).to.deep.equal([
            {
              message: row.error,
              range: new vscode.Range(
                row.lineNumber - 1,
                row.columnNumber - 1,
                row.lineNumber - 1,
                row.columnNumber - 1
              ),
              severity: vscode.DiagnosticSeverity.Error,
              source: row.type
            }
          ]);
``` lines 457-476

It separates the the assertions so we don't have to run it to know which on is failing.  A very uningration solution as it pulls the utility into the test, but it was the best path I could come up with.

Reply:
I updated the var names in this method to more closely reflect the params of the method where they are defined.